finagle-core/src/main/scala/com/twitter/finagle/Filter.scala

```diff
@@ -48,28 +50,9 @@ abstract class Filter[-ReqIn, +RepOut, +ReqOut, -RepIn]
    *
    * @param next another filter to follow after this one
    */
-  def andThen[Req2, Rep2](next: Filter[ReqOut, RepIn, Req2, Rep2]) =
-    new Filter[ReqIn, RepOut, Req2, Rep2] {
-      def apply(request: ReqIn, service: Service[Req2, Rep2]) = {
-        val svc: Service[ReqOut, RepIn] = new Service[ReqOut, RepIn] with Proxy {
-          // note that while `Service.rescue` could be used here it would
-          // entail an extra allocation.
-          def apply(request: ReqOut): Future[RepIn] = {
-            try {
-              next(request, service)
-            } catch {
-              case NonFatal(e) => Future.exception(e)
-            }
-          }
-          def self = service
-          override def close(deadline: Time) = service.close(deadline)
-          override def status = service.status
-          override def toString() = service.toString()
-        }
-
-        Filter.this.apply(request, svc)
-      }
-    }
+  def andThen[Req2, Rep2](next: Filter[ReqOut, RepIn, Req2, Rep2]): Filter[ReqIn, RepOut, Req2, Rep2] =
+    if (next eq Filter.identity) this.asInstanceOf[Filter[ReqIn, RepOut, Req2, Rep2]]
```

Comment:
> how frequently is identity used?

this will be difficult for me to determine. would you rather i left it out? it supports this neat equality:

    filter andThen identity eq filter
    (identity andThen filter) eq filter